forge-ai/src/main/java/forge/ai/ComputerUtilMana.java

```diff
@@ -666,47 +666,42 @@ private static boolean payManaCost(final ManaCostBeingPaid cost, final SpellAbil
 
         List<Mana> manaSpentToPay = test ? new ArrayList<>() : sa.getPayingMana();
         List<SpellAbility> paymentList = Lists.newArrayList();
+        final ManaPool manapool = ai.getManaPool();
 
-        if (ManaPool.payManaCostFromPool(cost, sa, ai, test, manaSpentToPay)) {
-            return true;    // paid all from floating mana
-        }
-
-        boolean purePhyrexian = cost.containsOnlyPhyrexianMana();
-
+        // Apply the color/type conversion matrix if necessary
+        manapool.restoreColorReplacements();
         boolean ignoreColor = false, ignoreType = false;
-        StaticAbility mayPlay = sa.getMayPlay();
+        CardPlayOption mayPlay = sa.getHostCard().mayPlay(sa.getMayPlay());
         if (mayPlay != null) {
-            if (mayPlay.hasParam("MayPlayIgnoreColor")) {
-                ignoreColor = true;
-            } else if (mayPlay.hasParam("MayPlayIgnoreType")) {
+            if (mayPlay.isIgnoreManaCostType()) {
                 ignoreType = true;
+            } else if (mayPlay.isIgnoreManaCostColor()) {
+                ignoreColor = true;
             }
-        } else if (sa.hasParam("ActivateIgnoreColor")) {
+            mayPlay.applyManaConvert(manapool);
+        } else if (sa.isActivatedAbility() && sa.getGrantorStatic() != null && sa.getGrantorStatic().hasParam("ManaConversion")) {
+            AbilityUtils.applyManaColorConversion(manapool, sa.getGrantorStatic().getParam("ManaConversion"));
             ignoreColor = true;
         }
+        StaticAbilityManaConvert.manaConvert(manapool, ai, sa.getHostCard(), sa);
```

Comment:
for these we need to check if effect is false (and mayPlay should probably check for isSpell too)

the last line with `StaticAbilityManaConvert.manaConvert(manapool, ai, sa.getHostCard(), sa);` should use `null` if effect is true